### ISSUE TITLE
Fields inside Row field can't be updated via UPDATE api and don't save in admin

### ISSUE THREAD

**Issue by 383bd03d** [payloadcms/payload#949]
This also don't create needed indexes in mongodb collection
Payload v. 1.0.7 - 1.0.24

**Comment by 383bd03d** [comment]
It happens only when i define custom id field

**Comment by DanRibbens** [comment]
I verified this is a bug.

This is the line causing the issue:
https://github.com/payloadcms/payload/blob/5825d7e64f234e8c644dc7962dbf1ca7a4a8ee64/src/mongoose/buildSchema.ts#L57

Because rows don't have a name, they are getting removed altogether from the schema when custom IDs are used.

The fix could be to change to this:
```js
const idIndex = schemaFields.findIndex((field) => (fieldAffectsData(field) && field.name === 'id'));
schemaFields.splice(idIndex, 1);
```

Although, when I tried to do that and for some reason I had other issues preventing save on the collection with the error relating to the ID field.
